src/main/java/ticTacToe/game/GameFlow.java

```diff
@@ -36,18 +38,16 @@ public void runGame() {
     }
 
     private void gameFlow(String currentMark) {
-        while (!grid.allOccupiedCells()) {
+        while (!gameEnded()) {
```

Comment:
👍

Reply:
This was a nice refactor Gabi. I like it because the 'levels of abstraction' are now equal. Ie,  the method names are all high level, so I can read the method names and understand what the code is doing. This means I only need to dig down into the lower level methods if I need to find out those details.